x-pack/legacy/plugins/siem/public/pages/detection_engine/rules/all_rules/batch_actions.tsx

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { EuiContextMenuItem } from '@elastic/eui';
+import React from 'react';
+import * as i18n from '../translations';
+import { TableData } from '../types';
+import { Action } from './reducer';
+import { deleteRulesAction, enableRulesAction, exportRulesAction } from './actions';
+
+export const getBatchItems = (
+  selectedState: TableData[],
+  dispatch: React.Dispatch<Action>,
+  closePopover: () => void,
+  kbnVersion: string
+) => {
+  const containsEnabled = selectedState.some(v => v.activate);
+  const containsDisabled = selectedState.some(v => !v.activate);
+  const containsLoading = selectedState.some(v => v.isLoading);
+
+  return [
+    <EuiContextMenuItem
+      key={i18n.BATCH_ACTION_ACTIVATE_SELECTED}
+      icon="checkInCircleFilled"
+      disabled={containsLoading || !containsDisabled}
+      onClick={async () => {
+        closePopover();
+        const deactivatedIds = selectedState.filter(s => !s.activate).map(s => s.id);
+        await enableRulesAction(deactivatedIds, true, dispatch, kbnVersion);
+      }}
+    >
+      {i18n.BATCH_ACTION_ACTIVATE_SELECTED}
+    </EuiContextMenuItem>,
+    <EuiContextMenuItem
+      key={i18n.BATCH_ACTION_DEACTIVATE_SELECTED}
+      icon="crossInACircleFilled"
+      disabled={containsLoading || !containsEnabled}
+      onClick={async () => {
+        closePopover();
+        const activatedIds = selectedState.filter(s => s.activate).map(s => s.id);
+        await enableRulesAction(activatedIds, false, dispatch, kbnVersion);
+      }}
+    >
+      {i18n.BATCH_ACTION_DEACTIVATE_SELECTED}
+    </EuiContextMenuItem>,
+    <EuiContextMenuItem
+      key={i18n.BATCH_ACTION_EXPORT_SELECTED}
+      icon="exportAction"
+      disabled={containsLoading || selectedState.length === 0}
+      onClick={async () => {
+        closePopover();
+        await exportRulesAction(
+          selectedState.map(s => s.sourceRule),
+          dispatch
+        );
+      }}
+    >
+      {i18n.BATCH_ACTION_EXPORT_SELECTED}
+    </EuiContextMenuItem>,
+    <EuiContextMenuItem
+      key={i18n.BATCH_ACTION_EDIT_INDEX_PATTERNS}
+      icon="indexEdit"
+      disabled={true}
+      onClick={async () => {
+        closePopover();
+      }}
+    >
+      {i18n.BATCH_ACTION_EDIT_INDEX_PATTERNS}
+    </EuiContextMenuItem>,
+    <EuiContextMenuItem
+      key={i18n.BATCH_ACTION_DELETE_SELECTED}
+      icon="trash"
+      disabled={containsLoading || selectedState.length === 0}
+      onClick={async () => {
+        closePopover();
+        await deleteRulesAction(
+          selectedState.map(r => r.sourceRule.id),
```

Comment:
These single letters throws me off a bit. Is the `r` rule here or was it suppose to be `s` as in `state`?

General musing, I usually use full words for as much as I can as the single letters don't convey much meaning.

Reply:
I would do something like this:
```ts
selectedStates.map(selectedState => selectedState.sourceRule.id)
```

or just destructure it like so if the variable name is in the way?:

```ts
selectedState.map(({ sourceRule }) => sourceRule.id)
```

Reply:
For inline `map`'s like this I try to use a single letter. Using `r` was a refactor flub though, so that should be `s` as you mention.

For more involved data transformations, I prefer using the full names like you said -- makes it clear what is being acted on.

For this though, I think object destructing is the way to go! Nice suggestion! We can go even further and do nested destructing ala:
``` ts 
selectedState.map(({ sourceRule: { id } }) => id),
```  

Thanks for the suggestion! 🙂